File: src/main/java/gargoyle/netsquares/store/prop/PrefStringProperty.java
package gargoyle.netsquares.store.prop;

public class PrefStringProperty extends BasePrefProperty<String> {
    public PrefStringProperty(Class<?> pkg, String key, String defaultValue) {
        super(pkg, key, defaultValue);
    }

    @Override
    protected void _set(String value) {
        preferences.put(key, value);
    }

    public String get(String defaultValue) {
        return preferences.get(key, defaultValue);
    }
}

File: src/main/kotlin/gargoyle/netsquares/store/prop/PrefStringProperty.kt
package gargoyle.netsquares.store.prop

import kotlin.reflect.KClass

class PrefStringProperty(pkg: KClass<*>, key: String, defaultValue: String) :
    BasePrefProperty<String>(pkg, key, defaultValue) {
    override fun _set(value: String) {
        preferences.put(key, value)
    }

    override fun get(defaultValue: String): String {
        return preferences[key, defaultValue]
    }
}